packages/core/src/utils/client.ts

```diff
@@ -39,7 +39,20 @@ export class Client {
     this.triggerEvent();
 
     if (autoConnect) {
-      setTimeout(async () => this.autoConnect(), 0);
+      let x = 0;
+      const intervalID = setInterval(() => {
+        let isReady = false;
+        for (const connector of connectors) {
+          if (connector.getProvider()) {
+            isReady = true;
+          }
+        }
+
+        if (++x === 5 || isReady) {
+          setTimeout(async () => this.autoConnect(), 0);
+          window.clearInterval(intervalID);
+        }
+      }, 1000);
```

Comment:
Please add more comments directly to code explaining what we're trying here. 
Later we can come back and improve.